### ISSUE TITLE
`message` and `details` fields are the same when you specify an `errmessage` for a constraint.

### ISSUE THREAD

**Issue by swaroopsm** [edgedb/edgedb#6768]
- EdgeDB Version: 4.4+aa576e1
- EdgeDB CLI Version:  4.0.2+500be79
- OS Version: macOS ventura

Steps to Reproduce:

1. For one of the properties in your schema add a `errmessage`, to customize the default errors emiitted for constraint validation.
2. Try to insert a record using the edgedb repl that fails this constraint
3. It returns the same error for `message` as well as `detail`

<img width="616" alt="Screenshot 2024-02-01 at 17 34 03" src="https://github.com/edgedb/edgedb/assets/898602/8acba0b9-8b50-4a4d-b348-4fd14e93de79">

Schema:
```
  type User extending Model {
    first_name: str;
    last_name: str;
    required email: str {
      constraint exclusive {
        errmessage := 'non-unique email'
      };
      constraint regexp(r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$') {
        errmessage := 'invalid email, try again later'
      };
    };
    is_email_verified: bool {
      default := false;
    };
  }
```

Originally asked here:
https://discord.com/channels/841451783728529451/849374705210490900/1202341124673634304

**Comment by aljazerzen** [comment]
The difference between values returned in error fields `message` and `details` is that `message` contains short names and `details` contains verbose names:

```
message: name violates exclusivity constraint
details: property 'name' of object type 'default::AbstractConstraintPureChild' violates exclusivity constraint
```

Do we want to make `message` contain the verbose names and remove `details`?

**Comment by aljazerzen** [comment]
The rationale for two fields is that:
- message is the thing that can trickle down to the end user (it can be shown in the UI of the developed app)
- details is the explanation of the error *for the developer* of the app.

This means that error `details` should not be affected by `errmessage` from the schema: it should still try to explain the constraint itself.

TLDR; the bug here is that details should not be `'invalid email, try again later'`, but `property 'name' of object type 'default::AbstractConstraintPureChild' violates exclusivity constraint`.
